### PR TITLE
Remove absl dependency

### PR DESCRIPTION
**Pull request by asuhan**
Re-implemented the minimal functionality we were using in a few separate commits.

### CODE REVIEW

File: lazy_tensor_core/lazy_tensors/int128.h [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+inline uint128 operator>>(uint128 lhs, int amount) {
+  return static_cast<unsigned __int128>(lhs) >> amount;
+}
+
+inline uint128 operator+(uint128 lhs, uint128 rhs) {
+  uint128 result = MakeUint128(Uint128High64(lhs) + Uint128High64(rhs),
+                               Uint128Low64(lhs) + Uint128Low64(rhs));
+  if (Uint128Low64(result) < Uint128Low64(lhs)) {  // check for carry
+    return MakeUint128(Uint128High64(result) + 1, Uint128Low64(result));
```
Comment (wconstab) [inline lazy_tensor_core/lazy_tensors/int128.h:91]: no check for overflow off high word?

**Review by wconstab — APPROVED** [review]
as discussed, it looks good for now, and we can discuss later on if we want to move the utilities somewhere else in the torch repo.

File: lazy_tensor_core/lazy_tensors/int128.h (hunk 1 above)
Comment (asuhan) [reply lazy_tensor_core/lazy_tensors/int128.h:91]: Nope, this is C++!

File: lazy_tensor_core/lazy_tensors/str_join.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+#include <algorithm>
+#include <iterator>
+#include <sstream>
+#include <string>
+#include <vector>
+
+namespace lazy_tensors {
+
+template <typename Range>
+std::string StrJoin(const Range& range, c10::string_view separator) {
```
[the remark this answers is not in the document]
Comment (asuhan) [reply lazy_tensor_core/lazy_tensors/str_join.h:14]: Yeah, let's land it as-is for now since we have adequate coverage from the rest of the tests anyway.

**Pull request merged**